core/workspace_comment/workspace_comment.js

```diff
@@ -163,3 +179,60 @@ Blockly.WorkspaceComment.prototype.setContent = function(content) {
     this.text_ = content;
   }
 };
+
+/**
+ * Encode a comment subtree as XML with XY coordinates.
+ * @param {boolean} opt_noId True if the encoder should skip the comment id.
+ * @return {!Element} Tree of XML elements.
+ */
+Blockly.WorkspaceComment.prototype.toXmlWithXY = function(opt_noId) {
+  var width;  // Not used in LTR.
+  if (this.workspace.RTL) {
+    // Here be performance dragons: On a rendered workspace, this will call
+    // getMetrics().
+    width = this.workspace.getWidth();
+  }
+  var element = this.toXml(opt_noId);
+  var xy = this.getRelativeToSurfaceXY();
+  element.setAttribute('x',
+      Math.round(this.workspace.RTL ? width - xy.x : xy.x));
+  element.setAttribute('y', Math.round(xy.y));
+  return element;
+};
+
+/**
+ * Encode a comment subtree as XML, but don't serialize the XY coordinates.
+ * This method avoids some expensive metrics-related calls that are made in
+ * toXmlWithXY().
+ * @param {boolean} opt_noId True if the encoder should skip the comment id.
+ * @return {!Element} Tree of XML elements.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.toXml = function(opt_noId) {
+  var commentElement = goog.dom.createDom('comment');
+  commentElement.setAttribute('h', this.getHeight());
+  commentElement.setAttribute('w', this.getWidth());
+  if (!opt_noId) {
+    commentElement.setAttribute('id', this.id);
+  }
+  commentElement.textContent = this.getContent();
+  return commentElement;
+};
+
+/**
+ * Decode an XML comment tag and create a comment on the workspace.
+ * @param {!Element} xmlComment XML comment element.
+ * @param {!Blockly.Workspace} workspace The workspace.
+ * @return {!Blockly.WorkspaceComment} The workspace comment created.
```

Comment:
Grammar nit, should be "The created workspace comment."

Reply:
Done.